torch_tvm/custom_tvm_ops/custom_dense.py

```diff
@@ -0,0 +1,79 @@
+import tvm
+from tvm import autotvm
+import topi
+from topi import nn
+from topi.util import get_const_tuple
+from topi.x86.dense import _declaration_dense_reshape, \
+        _declaration_dense_nopack_reshape, \
+        _declaration_dense_nopack, \
+        _declaration_dense_pack, \
+        _default_dense_nopack_config
+
+@autotvm.register_topi_compute(nn.dense, "cpu", "direct", override=True)
+def declare_customized_dense(cfg, data, weight, bias=None, out_dtype=None):
+    if len(data.shape) == 2:
+        batch, _ = get_const_tuple(data.shape)
+    elif len(data.shape) > 2:
+        batch = get_const_tuple(data.shape)[0]
+        pack_weights = (batch > 16)
+        return _declaration_dense_reshape(cfg, data, weight, \
+                bias, out_dtype, pack_weights)
+
+    # For small batch sizes, don't pack weight into cache-friendly layout
+    # because of overhead in packing and limited reuse from batch dimension
+    # TODO(icemelon9): use a more systematic way to determine which schedule to use
+    if batch <= 16:
+        return _declaration_dense_nopack(cfg, data, weight, bias, out_dtype)
+    return _declaration_dense_pack(cfg, data, weight, bias, out_dtype)
+
+@autotvm.register_topi_compute(nn.dense, "cpu", "custom_direct_reshape")
```

Comment:
I don't think we need this decorator.

Reply:
No? Do you mean just use `tvm.register_topi_compute`? I am not sure but the function does define split on the axis which can be used for tuning.